sensor/kubernetes/clustermetrics/cluster_metrics.go

```diff
@@ -30,13 +30,20 @@ type ClusterMetrics interface {
 }
 
 // New returns a new cluster metrics Sensor component.
-func New(k8sClient kubernetes.Interface) ClusterMetrics {
+func New(k8sClient kubernetes.Interface, pollInterval time.Duration) ClusterMetrics {
+	interval := pollInterval
+	if interval == 0 {
```

Comment:
Rather than having a default value for interval that users have to know, why not:
```go
func NewWithInterval(k8sClient kubernetes.Interface, pollInterval time.Duration) ClusterMetrics {
 // ...
}

func New(k8sClient kubernetes.Interface) ClusterMetrics {
        return NewWithInterval(k8sClient, defaultInterval)
}
```

Reply:
Good idea. I've added it in [5e44505](https://github.com/stackrox/stackrox/pull/7183/commits/5e44505bbe71d5284dc107000002a85e929beff7)